docker/Dockerfile

```diff
@@ -218,7 +221,7 @@ RUN export TRILINOS_HASH=7b530eda6fdda0a4f17d192dfe7c4e07d582110e && \
       -D Kokkos_ENABLE_Cuda_Lambda=ON \
       -D Kokkos_ARCH_VOLTA70=ON \
       -D Tpetra_INST_CUDA=ON \
-      -D Trilinos_CXX11_FLAGS="-std=c++11 --expt-extended-lambda" \
+      -D Trilinos_CXX11_FLAGS="-std=c++14 --expt-extended-lambda" \
```

Comment:
Please explain why this is needed in this PR.

Reply:
Ah, good point. This should not have any effect any more, see https://github.com/trilinos/Trilinos/pull/6814.

Reply:
While this is a helpful reference, the main thrust of my question was why this PR changes the standard from 11 to 14 without a need? As it does not update ArborX, it seems that such a such is unwarranted here if it worked with 11 before.

Reply:
My thinking is that if we want C++14 anyway it's a good idea to also tell Kokkos and whatever else in Trilinos would do something else for C++14. I don't want to fight over this at all, though. If you prefer I can just drop specifying any standard at all or even avoid updating Trilinos.

Reply:
I'm not opposed to it, just wanted to understand if this change is required or not.